pkg/controller/master/upgradelog/controller.go

```diff
@@ -121,52 +136,51 @@ func (h *handler) OnUpgradeLogChange(_ string, upgradeLog *harvesterv1.UpgradeLo
 			logrus.Info("rancher-logging Addon is not installed")
 		} else {
 			if addon.Spec.Enabled {
+				// mark the operator source
+				setLoggingOperatorSource(toUpdate, util.RancherLoggingName)
 				setOperatorDeployedCondition(toUpdate, corev1.ConditionTrue, "Skipped", "rancher-logging Addon is enabled")
 				return h.upgradeLogClient.Update(toUpdate)
 			}
 			logrus.Info("rancher-logging Addon is not enabled")
 		}
 
-		// Detect the rancher-logging ManagedChart
-		managedChart, err := h.managedChartCache.Get(util.FleetLocalNamespaceName, util.RancherLoggingName)
-		if err != nil {
-			if !apierrors.IsNotFound(err) {
-				return nil, err
-			}
-			logrus.Info("rancher-logging ManagedChart is not installed")
-		} else {
-			if managedChart.Status.Summary.DesiredReady > 0 && managedChart.Status.Summary.DesiredReady == managedChart.Status.Summary.Ready {
-				setOperatorDeployedCondition(toUpdate, corev1.ConditionTrue, "Skipped", "rancher-logging ManagedChart is ready")
-				return h.upgradeLogClient.Update(toUpdate)
-			}
-			logrus.Warn("rancher-logging ManagedChart is not ready")
-			return nil, err
-		}
-
-		// If none of the above exists, install the customized rancher-logging ManagedChart
-		logrus.Info("Deploy logging-operator")
+		logrus.Info("Deploy logging-operator via a new ManagedChart")
 		if _, err := h.managedChartClient.Create(prepareOperator(upgradeLog)); err != nil && !apierrors.IsAlreadyExists(err) {
 			return nil, err
 		}
+		// format: hvst-upgrade-l5875-upgradelog-operator
+		setLoggingOperatorSource(toUpdate, name.SafeConcatName(upgradeLog.Name, util.UpgradeLogOperatorComponent))
 
 		return h.upgradeLogClient.Update(toUpdate)
 	}
 
 	// Try to establish the logging infrastructure by creating a customized Logging resource
 	if harvesterv1.LoggingOperatorDeployed.IsTrue(upgradeLog) && harvesterv1.InfraReady.GetStatus(upgradeLog) == "" {
-		logrus.Info("Start to create the logging infrastructure for the upgrade procedure")
+		logrus.Info("Start to create the logging infrastructure and fluentbitagent for the upgrade procedure")
 
 		toUpdate := upgradeLog.DeepCopy()
 
 		// The creation of the Logging resource will indirectly bring up fluent-bit DaemonSet and fluentd StatefulSet
-		candidateImages, err := h.getConsolidatedLoggingImageList(name.SafeConcatName(upgradeLog.Name, util.UpgradeLogOperatorComponent))
+		//candidateImages, err := h.getConsolidatedLoggingImageList(name.SafeConcatName(upgradeLog.Name, util.UpgradeLogOperatorComponent))
+		// TODO: need to identify which source to get images, currently it only gets from rancher-logging addons, assue addon is enabled
```

Comment:
Nit:

> ... , assue addon is enabled

*assume ?

Reply:
ah, I need to remove the `TODO` comments, it was resolved later ...